portal/src/main/webapp/js/src/cross-cancer-plotly-plots.js

```diff
@@ -515,80 +520,97 @@ var ccPlots = (function (Plotly, _, $) {
-                            legend_str = legend_str.substring(legend_str.indexOf(">") + 1, legend_str.length);
-                            var final_pdf_str = main_plots_str + legend_str;
-
-                            final_pdf_str = final_pdf_str.replace(/"/g, "'");
-                            final_pdf_str = final_pdf_str.replace("xmlns='http://www.w3.org/2000/svg' xmlns:xlink='http://www.w3.org/1999/xlink'", "");
-                            final_pdf_str = final_pdf_str.replace(/text-anchor='end'/g, "");
-                            final_pdf_str = final_pdf_str.replace(/text-anchor='start'/g, "");
-                            final_pdf_str = final_pdf_str.replace(/text-anchor='middle'/g, "");
-                            final_pdf_str = final_pdf_str.replace(/text-anchor: start;/g, "");
-                            final_pdf_str = final_pdf_str.replace(/font-family: 'Open Sans',/g, "");
-                            final_pdf_str = final_pdf_str.replace(/fill: transparent;/g, "fill-opacity: 0;");
-                            final_pdf_str = final_pdf_str.replace(/'/g, "\"");
-
-                            var downloadOptions = {
-                                filename: "cross-cancer-plots.pdf",
-                                contentType: "application/pdf",
-                                servletName: "svgtopdf.do"
-                            };
-                            cbio.download.initDownload(final_pdf_str, downloadOptions);
 
-                        });
-                        $("#cc_plots_data_download").click(function() {
-                            var get_tab_delimited_data = function() {
-                                var result_str = "Sample Id" + "\t" + "Cancer Study" + "\t" + "Profile Name" + "\t" + "Gene" + "\t" + "Mutation" + "\t" + "Value" + "\n";
-                                _.each(_d[0].concat(_d[1], _d[2]), function(_obj) {
-                                    if ( _obj.sequenced) {
-                                        if (_obj.mutation_type === "non" ) {
-                                            result_str += _obj.sample_id + "\t" + _obj.study_name + "\t" + "RNA Seq V2" + "\t" + gene[0] + "\t" + "Not Mutated" + "\t" + _obj.profile_data + "\n";
+                    $.when(ccPlots.util.getGeneticProfiles_(_.pluck(_.pluck(window.studies.models, "attributes"), "studyId"))).then(function(_profiles) {
+                        
+                        // if no v2 profile (data) available, set profile selection to v1
+                        if($("#cc_plots_profile_list option").length === 0) {
+                            var _hasV2 = false;
+                            _.each(_.pluck(_profiles, "id"), function(_pid) {
+                                if (_pid.indexOf("rna_seq_v2_mrna") !== -1) _hasV2 = true;
+                            });
+                            if (!_hasV2) $("#cc_plots_profile_list").append("<option value='v1'>RNA Seq</option><option value='v2'>RNA Seq V2</option>");
```

Comment:
It's better to show a message saying we don't have data for that profile that just removing the option quietly.